packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -181,46 +194,56 @@ class CryptoWatchPriceFeed extends PriceFeedInterface {
 
     // Round down to the nearest ohlc period so the queries captures the OHLC of the period *before* this earliest
     // timestamp (because the close of that OHLC may be relevant).
-    const earliestHistoricalTimestamp = Math.floor((currentTime - this.lookback) / this.ohlcPeriod) * this.ohlcPeriod;
+    const earliestHistoricalTimestamp =
+      Math.floor((currentTime - (this.lookback + this.twapLength)) / this.ohlcPeriod) * this.ohlcPeriod;
```

Comment:
probably wrong, but this logic is confusing. i can imagine though if you used `Math.ceil` instead, you would get a longer lookback time because of the multiplication by `ohlcPeriod` at the end, adding an additional ohlc period if there is a decimal, rather than truncating it.  Not totally sure though if that is the intent

Reply:
yeah, this is a little confusing. The floor makes the lookback longer because we're not computing a distance or difference, we're computing the _actual timestamp_. We would ceil the _distance_ or _difference_, but to create a timestamp that is further in the past, we want to floor since that creates the lower value. Does that makes sense?

Reply:
yea that makes sense, we want the smaller timestamp. for some reason i kept thinking it was time ago ( where larger number is further in past), rather than absolute time in past.